langchain/client/runner_utils.py

```diff
@@ -57,6 +57,33 @@ class InputFormatError(Exception):
 ## Shared Utilities
 
 
+def _wrap_in_chain_factory(
+    llm_or_chain_factory: Union[Chain, MODEL_OR_CHAIN_FACTORY],
+    dataset_name: str = "<my_dataset>",
+) -> MODEL_OR_CHAIN_FACTORY:
+    """Forgive the user if they pass in a chain without memory instead of a chain
```

Comment:
this makes sense if memory is the ONLY possible state that is stored by chains. there might be a good reason to use factories for all chains that we're not thinking of.

Reply:
yeah true...